models.py

```diff
@@ -86,42 +87,57 @@ def lp(self, params):
         lp = 0
         mu_index = self.param_index_dict["mu"]
         tau_index = self.param_index_dict["tau"]
-        theta_indexes = np.array(self.param_index_dict["theta_trans"])
-        theta = params[mu_index] + params[theta_indexes] * params[tau_index]
+        theta_t_indexes = np.array(self.param_index_dict["theta_t"])
+        theta = params[mu_index] + params[theta_t_indexes] * params[tau_index]
         # theta is length (n_schools)
-        lp += np.sum(logpdf(params[theta_indexes], 0, 1))
+        lp += np.sum(logpdf(params[theta_t_indexes], 0, 1))
+        lp += np.sum(logpdf(self.effects, theta, self.sigma))
+        return lp
+
+    def unconstrain_lp(self, params):
+        lp = 0
+        mu_index = self.param_index_dict["mu"]
+        logtau_index = self.param_index_dict["tau"]
+        theta_t_indexes = np.array(self.param_index_dict["theta_t"])
+        theta = params[mu_index] + params[theta_t_indexes] * np.exp(params[logtau_index])
+        lp += params[logtau_index] #jacobian adjustment for log transform
+        lp += np.sum(logpdf(params[theta_t_indexes], 0, 1))
         lp += np.sum(logpdf(self.effects, theta, self.sigma))
         return lp
 
     def lp_sum(self, params):
         lp = 0
         mu_index = self.param_index_dict["mu"]
-        tau_index = self.param_index_dict["tau"]
-        theta_indexes = np.array(self.param_index_dict["theta_trans"])
+        logtau_index = self.param_index_dict["tau"]
+        theta_t_indexes = np.array(self.param_index_dict["theta_t"])
         theta = np.tile(params[:, mu_index], (self.school_count, 1)).T + \
-                params[:, theta_indexes] * np.tile(params[:, tau_index], (self.school_count, 1)).T
+                params[:, theta_t_indexes] * np.tile(np.exp(params[:, logtau_index]), (self.school_count, 1)).T
         # theta is dim (n_draws, n_schools)
-        lp += np.sum(logpdf(params[:, theta_indexes], 0, 1))
+        lp += np.sum(params[logtau_index]) * params.shape[0] #TODO check * params.shape[0] (= n_draws)
```

Comment:
as we are computing lp for `n_draws` number of samples at the same time (for faster computation), thought corresponding `lp` should be added  `n_draws ` = ` params.shape[0]` times. need checking @Dashadower

Reply:
If `params` was an array with dimensions `(n_sample, n_params)`, and we wish to sum up `logtau_index` for all samples(rows), we would index it by:
`lp += np.sum(params[:, logtau_index])`
Then np.sum will sum up the indexed vector of logtaus.

That said, I don't think this function is necessary, since it's only used in `vi()` for printing to output. We can change `elbo_grad()` in NormalMeanField to return both elbo and grad, since the function is currently calculating elbo values but trashing them.